shared/types/background.ts

```diff
@@ -0,0 +1,376 @@
+/* eslint-disable import/no-restricted-paths */
+/** Circular dependencies in this file should have no impact at build time and runtime, ås all imports/exports are types that will be stripped by tsc */
```

Comment:
We _should_ be ignoring `import type` style imports when checking for `circular dependencies`. Is that not the case?

Reply:
This was more of an explanation for ignoring `import/no-restricted-paths`, which still finds violations in this file despite all imports being `import type`. 

This might be clearer:

```suggestion
/** Type import paths do not need to be restricted, as they are stripped at runtime and do not have any build time impact. */
```